app/src/main/java/net/grandcentrix/blutufintegration/domain/ScanUseCase.kt

```diff
@@ -1,12 +1,14 @@
 package net.grandcentrix.blutufintegration.domain
 
+import kotlinx.coroutines.flow.SharedFlow
+import net.grandcentrix.blutufintegration.data.model.DeviceUiState
+import net.grandcentrix.blutufintegration.data.model.ProcessState
 import net.grandcentrix.blutufintegration.data.repo.BluetoothRepository
 
 class ScanUseCase(private val bluetoothRepository: BluetoothRepository) {
 
-    val devicesFlow = bluetoothRepository.devicesStateFlow
-
-    suspend fun execute() {
+    suspend fun execute() : SharedFlow<ProcessState<List<DeviceUiState>>> {
```

Comment:
In an ideal world the `BluetoothRepository` would expose a `Flow` while using a `SharedFlow` under the hood. But that is not the scope of this PR.
Please change the return type of `execute()` to `Flow` anyways.

Reply:
👍👍